libarchive/entry.py

```diff
@@ -61,17 +61,36 @@ def isfifo(self):
 
     @property
     def islnk(self):
+        return bool(ffi.entry_hardlink_w(self._entry_p))
+
+    @property
+    def issym(self):
         return self.filetype & 0o170000 == 0o120000
 
     @property
+    def linkpath(self):
+        if self.islnk:
+            return ffi.entry_hardlink_w(self._entry_p)
+        if self.issym:
+            return ffi.entry_symlink_w(self._entry_p)
```

Comment:
@Changaco The use case for linkpath is to have the same api as tarfile that uses a single linkpath for either hard of sym links, except that tarfile does not use properties by methods, at least keep the same names.